### PR TITLE
feat: v11

### LINKED ISSUE #589: Only accept string payloads for `webhooks.verify()` and `webhooks.verifyAndReceive()`
**What’s missing?**

We have recurring problems with some event payloads where the code returns a "signature does not match" error, mostly reported by Probot users who are probably the biggest share of users of `@octokit/webhooks`.

I think we should stop making assumptions about how a JSON payload is stringified by GitHub and only accept the raw request string, as all other webhooks SDKs do that I know of, e.g. [`stripe`](https://github.com/stripe/stripe-node#webhook-signing)

I'd consider passing the raw request body string as a best practice today, and we enforce best practices in the @octokit modules.

The challenge that this will bring is that the raw request body is not always easily accessible by server frameworks or serverless environments, so we should document how to do it with e.g. `express`, AWS Lambda, Vercel, Begin, Azure Functions, Google Cloud Functions, Cloudflare Workers, and invite users to add examples for other platforms.

**Alternatives you tried**

n/a

### LINKED ISSUE #775: [BUG]: dependabot_alert events fail to verify
## Observed Behavior

I noticed that `webhooks.verify()` returns false for dependabot_alert events, but only for this event type. All other events verify successfully.

To reproduce, I created a small script:

```ts
import { Webhooks } from '@octokit/webhooks'

const webhooks = new Webhooks({ secret: 'redacted' })

const check_run_event = {
  sign: "sha256=redacted", // from the X-Hub-Signature-256 header of the check_run event
  data: { /* copied a check_run event payload from the App Settings */ }
}
const dependabot_alert_event = {
  sign: "sha256=redacted", // from the X-Hub-Signature-256 header of the dependabot_alert event
  data: { /* copied a dependabot_alert event payload from the App Settings */ }
}

await webhooks.verify(dependabot_alert_event.data, dependabot_alert_event.sign)
// -> false

await webhooks.verify(check_run_event.data, check_run_event.sign)
// -> true
```

## Expected Behavior

Given that other events verify successfully, I would've expected that the dependabot_alert event also verifies successfully.

## Note

I'm unsure if this is a bug of @octokit/webhooks, or maybe even a miscalculation of the `X-Hub-Signature-256` header coming from GitHub-Hookshot. In case it really seems to be an issue of Github-Hookshot, where can I report this?

I'm aware that the dependabot_alert event is still in beta.

### Versions

@octokit/webhooks v10.3.1

### Relevant log output

```shell
No logs are printed
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #828: Modify `package.json` fields for ESM and browser builds best practices
> Should we change the current value of the `module` field to point to `dist-src/` and add a new `browser` field that points to `dist-web/` ?
> 
> This would alleviate issues like [octokit/auth-app.js#59](https://github.com/octokit/auth-app.js/issues/59) until work on `octokit-next` is finished
> 
> ```diff
> diff --git a/scripts/build.mjs b/scripts/build.mjs
> index 3ef8688..73d9ef9 100644
> --- a/scripts/build.mjs
> +++ b/scripts/build.mjs
> @@ -72,9 +71,9 @@ async function main() {
>          ...pkg,
>          files: ["dist-*/**", "bin/**"],
>          main: "dist-node/index.js",
> -        module: "dist-web/index.js",
> +        browser: "dist-web/index.js",
>          types: "dist-src/index.d.ts",
> -        source: "dist-src/index.js",
> +        module: "dist-src/index.js",
>        },
> ```

I would do that in a separate PR (feel free to create a new issue mentioning this comment) so we do not add more things on top of this migration. I think it's good the differences are minimum in this PR. What do you think?

_Originally posted by @oscard0m in https://github.com/octokit/webhooks.js/pull/819#issuecomment-1498924723_

### PR DESCRIPTION
**Pull request by wolfy1339**
Fixes #589, fixes #775, fixes #828

### Breaking changes
* middleware only and (always) handles requests at `path`
* Only accept string payloads for `webhooks.verify()` and `webhooks.verifyAndReceive()` and `webhooks.sign()`
* The middleware now expects only string payloads for `request.body`
* Modifies the `module` field in `package.json` to point to the source ESM, and adds a `browser` field that points to the browser bundle

### CODE REVIEW

File: src/middleware/node/get-payload.ts [hunk 1]
```diff
@@ -12,20 +11,11 @@ import AggregateError from "aggregate-error";
...
 
-  if (request.body) {
-    if (typeof request.body !== "string") {
-      console.error(
-        "[@octokit/webhooks] Passing the payload as a JSON object in `request.body` is deprecated and will be removed in a future release of `@octokit/webhooks`, please pass it as a a `string` instead."
-      );
-    }
-    return Promise.resolve(request.body as WebhookEvent | string);
-  }
+  if (request.body) return Promise.resolve(request.body);
```
Comment (philipandersson) [inline src/middleware/node/get-payload.ts:18]: Finally no more objects 🙏

File: src/verify-and-receive.ts [hunk 2]
```diff
@@ -35,9 +29,6 @@ export async function verifyAndReceive(
   return state.eventHandler.receive({
     id: event.id,
     name: event.name,
-    payload:
-      typeof event.payload === "string"
-        ? JSON.parse(event.payload)
-        : event.payload,
+    payload: JSON.parse(event.payload),
```
Comment (philipandersson) [inline src/verify-and-receive.ts:32]: Great that we are addressing: https://github.com/octokit/webhooks.js/issues/589#issuecomment-1376814798
I guess you could argue that it should be the `receive` function's responsibility to parse the `JSON`.

**Review by oscard0m — DISMISSED** [review]

**Comment by wolfy1339** [comment]
@octokit/js @octokit/extensibility-sdks Is there anything else that you could think of that should be included in this major version

**Comment by oscard0m** [comment]
> @octokit/js @octokit/extensibility-sdks Is there anything else that you could think of that should be included in this major version

As you mentioned, I agree on adding this as well: https://github.com/octokit/webhooks.js/issues/828

**Comment by wolfy1339** [comment]
Once we land #819, I can work on addressing #828 in this PR

**Comment by wolfy1339** [comment]
I addressed #828 in ffbf0ac70ae8fb901029a169b1c677e6a55bef7d

**Review by kfcampbell — APPROVED** [review]
I'm :+1: on this, thank you @wolfy1339!

**Pull request merged**